utils/src/treev2/quickPickWizard/quickPickAzureResource/QuickPickAppResourceStep.ts

```diff
@@ -19,27 +20,33 @@ export class QuickPickAppResourceStep extends GenericQuickPickStep<ResourceGroup
         const pickedAppResource = await super.promptInternal(wizardContext) as AppResourceItem;
 
         // TODO
-        wizardContext.resource = pickedAppResource;
-        wizardContext.resourceGroup = pickedAppResource.resourceGroup;
+        wizardContext.resource = pickedAppResource.resource;
+        wizardContext.resourceGroup = pickedAppResource.resource.resourceGroup;
 
         return pickedAppResource;
     }
 
     protected isDirectPick(node: AppResourceItem): boolean {
+        // TODO do this before calling isDirectPick/isIndirectPick in GenericQuickPickStep?
+        node = isBox(node) ? node.unwrap() : node;

```

Comment:
Mentioned above but I think we shouldn't do any unboxing until the very end of the experience (else the TDP won't understand what objects we're giving it when we call `getChildren()`).